### ISSUE TITLE
UnresolvedFormat error when converting time intervals that span noon to strings

### ISSUE THREAD

**Issue by bryanlep** [elixir-cldr/cldr_dates_times#22]
Hello, I want to start by saying thanks for this awesome library!

I've encountered an issue after updating to version `2.7.1`.  Using `Cldr.Time.Interval.to_string/4` where the start time is before noon and the end time is after produces an error when it didn't before, when using `DateTime` structs. Here's a simple example:
```elixir
Cldr.Time.Interval.to_string(~U[2021-05-06 11:45:00Z], ~U[2021-05-06 12:15:00Z], MyApp.Cldr, format: :bhm)
```
produces:
```elixir
{:error,
 {Cldr.DateTime.UnresolvedFormat,
  "The interval format %{b: [\"h:mm B – \", \"h:mm B\"], h: [\"h:mm – \", \"h:mm B\"], m: [\"h:mm – \", \"h:mm B\"]} is invalid. Valid formats are [:long, :medium, :short] or an interval format string."}}
```

I have verified that version `2.7.0` does not emit this error and formats the string correctly.

**Comment by kipcole9** [comment]
@bryanlep, sorry for the inconvenience. Thats a very unpleasant regression. Will work on it and have a fix in the next 12 hours.

**Comment by kipcole9** [comment]
As of commit [1fee73]( https://github.com/elixir-cldr/cldr_dates_times/commit/1fee73933b9ccd74a64027046cffb30014369238) I believe the regression is fixed. I have published [ex_cldr_dates_times version 2.7.2](https://hex.pm/packages/ex_cldr_dates_times/2.7.2) with the changelog entry:

### Bug Fixes

* Fix regression time intervals where the start time is "a.m." and the end time is "p.m." and the format code is `:b` (previously assumed `:b`). Thanks to @bryanlep for the report. Closes #22.

Apologies again for the regression, thats for the report and collaboration.

**Comment by bryanlep** [comment]
No worries, thanks for the quick fix!
